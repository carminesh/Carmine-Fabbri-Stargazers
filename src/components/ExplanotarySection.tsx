import { StyleSheet, Text, View } from 'react-native';
import React, { memo } from 'react';
import { SvgXml } from 'react-native-svg';
import { searchStarredUsers } from '../assets/svg/searchStarredUsers';
import ColorsPalette from '../constants/ColorsPalette';

interface ExplanotarySectionProps {
    svgImage: string;
    textHeader: string;
    textSubheader: string;
}

const ExplanotarySection: React.FC<ExplanotarySectionProps> = ({ svgImage, textHeader, textSubheader }) => {
    return (
        <View style={styles.svgContainer}>
            <SvgXml width={'45%'} height={'45%'} xml={svgImage} />
            <Text style={styles.textHeader}>{textHeader}</Text>
            <Text style={styles.textSubheader}>{textSubheader}</Text>
        </View>
    );
};

export default memo(ExplanotarySection);

const styles = StyleSheet.create({
    svgContainer: {
        backgroundColor: ColorsPalette.WHITE90,
        flex: 1,
        height: '100%',
        width: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        paddingBottom: 24,
    },
    textHeader: {
        paddingTop: 14,
        fontWeight: 'bold',
        fontSize: 26,
        color: ColorsPalette.BLACK_ACCENT,
    },
    textSubheader: {
        paddingTop: 6,
        fontSize: 16,
        fontWeight: '400',
    },
});
