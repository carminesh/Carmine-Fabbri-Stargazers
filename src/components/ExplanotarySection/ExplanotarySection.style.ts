import { StyleSheet } from 'react-native';
import ColorsPalette from '../../constants/ColorsPalette';

export const styles = StyleSheet.create({
    svgContainer: {
        backgroundColor: ColorsPalette.WHITE90,
        flex: 1,
        height: '100%',
        width: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        paddingBottom: 24,
    },
    textHeader: {
        paddingTop: 14,
        fontWeight: 'bold',
        fontSize: 26,
        color: ColorsPalette.BLACK_ACCENT,
    },
    textSubheader: {
        paddingTop: 6,
        fontSize: 16,
        fontWeight: '400',
    },
});