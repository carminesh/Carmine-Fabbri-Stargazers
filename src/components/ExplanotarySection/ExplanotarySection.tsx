import { Text, View } from 'react-native';
import React, { memo } from 'react';
import { SvgXml } from 'react-native-svg';
import { styles } from './ExplanotarySection.style';

interface ExplanotarySectionProps {
    svgImage: string;
    textHeader: string;
    textSubheader: string;
}

const ExplanotarySection: React.FC<ExplanotarySectionProps> = ({ svgImage, textHeader, textSubheader }) => {
    return (
        <View style={styles.svgContainer}>
            <SvgXml testID="explanotary-section-svg" width={'45%'} height={'45%'} xml={svgImage} />
            <Text testID="Test Header" style={styles.textHeader}>
                {textHeader}
            </Text>
            <Text testID="Test Subheader" style={styles.textSubheader}>
                {textSubheader}
            </Text>
        </View>
    );
};

export default memo(ExplanotarySection);
