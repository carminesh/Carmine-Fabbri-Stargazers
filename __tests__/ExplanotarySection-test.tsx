import 'react-native';
import React from 'react';
import { render } from '@testing-library/react-native';
import ExplanotarySection from '../src/atoms/ExplanotarySection/ExplanotarySection';
import renderer from 'react-test-renderer';

describe('ExplanotarySection testing', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<ExplanotarySection svgImage="<svg>...</svg>" textHeader="Test Header" textSubheader="Test Subheader" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should display the svg image passed as a prop', async () => {
        const svgImage = '<svg>...</svg>';
        const { queryByTestId } = render(<ExplanotarySection svgImage={svgImage} textHeader="Test Header" textSubheader="Test Subheader" />);

        const svg = queryByTestId('explanotary-section-svg');
        expect(svg).toBeTruthy();
    });

    it('should display the svg and the text header passed as a prop', async () => {
        const svgImage = '<svg>...</svg>';
        const { getByText } = render(<ExplanotarySection svgImage={svgImage} textHeader="Test Header" textSubheader="Test Subheader" />);

        const textHeader = getByText('Test Header');
        expect(textHeader).toBeTruthy();
    });

    it('should display the svg and the text subheader passed as a prop', async () => {
        const svgImage = '<svg>...</svg>';
        const { getByText } = render(<ExplanotarySection svgImage={svgImage} textHeader="Test Header" textSubheader="Test Subheader" />);

        const textSubheader = getByText('Test Subheader');
        expect(textSubheader).toBeTruthy();
    });
});
