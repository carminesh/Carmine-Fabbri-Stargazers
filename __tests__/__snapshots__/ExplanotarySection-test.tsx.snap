// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExplanotarySection testing renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f2f2f7",
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "paddingBottom": 24,
      "width": "100%",
    }
  }
>
  <RNSVGSvgView
    bbHeight="45%"
    bbWidth="45%"
    focusable={false}
    height="45%"
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": "45%",
          "width": "45%",
        },
      ]
    }
    testID="explanotary-section-svg"
    width="45%"
    xml="<svg>...</svg>"
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    >
      ...
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      {
        "color": "#333333",
        "fontSize": 26,
        "fontWeight": "bold",
        "paddingTop": 14,
      }
    }
    testID="Test Header"
  >
    Test Header
  </Text>
  <Text
    style={
      {
        "fontSize": 16,
        "fontWeight": "400",
        "paddingTop": 6,
      }
    }
    testID="Test Subheader"
  >
    Test Subheader
  </Text>
</View>
`;
